Dockerfile

```diff
@@ -0,0 +1,22 @@
+#FROM python:3.10.11
+#FROM wallies/python-cuda:3.10-cuda11.6-runtime
+
+# Using argument for base image to avoid multiplying Dockerfiles
+ARG BASEIMAGE
+FROM $BASEIMAGE
+
+RUN groupadd -g 10009 -o privategpt && useradd -m -u 10009 -g 10009 -o -s /bin/bash privategpt
+USER privategpt
```

Comment:
Tried running ingest docker container in linux and getting this error:
```
$ sudo docker compose run --rm privategpt-ingest
[sudo] password for brian:
Loading documents from /home/privategpt/source_documents
Loaded 1 documents from /home/privategpt/source_documents
Split into 90 chunks of text (max. 500 characters each)
Traceback (most recent call last):
  File "/home/privategpt/src/ingest.py", line 97, in <module>
    main()
  File "/home/privategpt/src/ingest.py", line 88, in main
    embeddings = HuggingFaceEmbeddings(model_name=embeddings_model_name)
  File "/home/privategpt/.local/lib/python3.10/site-packages/langchain/embeddings/huggingface.py", line 54, in __init__
    self.client = sentence_transformers.SentenceTransformer(
  File "/home/privategpt/.local/lib/python3.10/site-packages/sentence_transformers/SentenceTransformer.py", line 87, in __init__
    snapshot_download(model_name_or_path,
  File "/home/privategpt/.local/lib/python3.10/site-packages/sentence_transformers/util.py", line 476, in snapshot_download
    os.makedirs(nested_dirname, exist_ok=True)
  File "/usr/local/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/usr/local/lib/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/privategpt/.cache/torch/sentence_transformers'
```
googling it suggests that its related to the Dockerfile USER not having correct permissions, but I'm not sure.

Do you know what could be causing this? I ran it fine on windows through wsl2 docker desktop, but get this error when running on a linux machine.

Reply:
It's related to the fact that you're running the docker container as root, and the unpriviliged container user can't create directories as root.

For this particular error `mkdir cache ; chmod 777 cache` should do the trick, you also need to do this for the 'db' directory.

Reply:
thank you, that fixed it. maybe worth noting in the readme? not sure how many other people will get this